### ISSUE TITLE
Don't require a `--files` flag if we're running only global bears

### ISSUE THREAD

**Issue by Makman2** [coala/coala#4928]

**Comment by newbazz** [comment]
running global bears means not specifing the `--bears` param?

**Comment by Makman2** [comment]
No. Some bears are so called `GlobalBear`s. They operate on the whole repo instead of individual files. However, some `GlobalBear`s don't process files at all (like the `GitCommitBear`), and those wouldn't need files to run.

**Comment by ishanSrt** [comment]
@Makman2 is this bear specific or absolutely all global bears don't require a `--files` flag

**Comment by ishanSrt** [comment]
@Makman2 I am assuming my previous comment, and I am opening a PR. Also assign this issue to me. If only global bears are being run and only the files argument is not provided, coala will run!

**Comment by Makman2** [comment]
> @Makman2 is this bear specific or absolutely all global bears don't require a --files flag

It's bear specific.

**Comment by Makman2** [comment]
Maybe we shouldn't require `--files` in general and assume "no files" by default? Not sure though...
